src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/convert/MappingDocumentDbConverter.java

```diff
@@ -81,43 +82,37 @@ public MappingDocumentDbConverter(
 
     @Override
     public void write(Object sourceEntity, Document document) {
-        if (sourceEntity == null) {
-            return;
-        }
-
-        final DocumentDbPersistentEntity<?> entity = mappingContext.getPersistentEntity(sourceEntity.getClass());
-        writeInternal(sourceEntity, document, entity);
+        // Not used anymore
+        throw new NotImplementedException();
     }
 
-    public void writeInternal(final Object entity,
-                              final Document targetDocument,
-                              final DocumentDbPersistentEntity<?> persistentEntity) {
-        if (entity == null) {
-            return;
+    public Document writeDoc(Object sourceEntity) {
+        if (sourceEntity == null) {
+            return null;
         }
 
-        if (persistentEntity == null) {
-            throw new MappingException("no mapping metadata for entity type: " + entity.getClass().getName());
-        }
+        try {
+            final DocumentDbPersistentEntity<?> persistentEntity =
+                    mappingContext.getPersistentEntity(sourceEntity.getClass());
 
-        final ConvertingPropertyAccessor accessor = getPropertyAccessor(entity);
-        final DocumentDbPersistentProperty idProperty = persistentEntity.getIdProperty();
+            if (persistentEntity == null) {
+                throw new MappingException("no mapping metadata for entity type: " + sourceEntity.getClass().getName());
+            }
 
-        if (idProperty != null) {
-            final Object value = accessor.getProperty(idProperty);
-            final String id = value == null ? null : value.toString();
-            targetDocument.setId(id);
-        }
+            final ConvertingPropertyAccessor accessor = getPropertyAccessor(sourceEntity);
+            final DocumentDbPersistentProperty idProperty = persistentEntity.getIdProperty();
 
-        for (final Field field : entity.getClass().getDeclaredFields()) {
-            if (idProperty != null && field.getName().equals(idProperty.getName())) {
-                continue;
-            }
+            final Document document = new Document(objectMapper.writeValueAsString(sourceEntity));
 
-            final PersistentProperty property = persistentEntity.getPersistentProperty(field.getName());
-            Assert.notNull(property, "Property is null.");
+            if (idProperty != null) {
+                final Object value = accessor.getProperty(idProperty);
+                final String id = value == null ? null : value.toString();
+                document.setId(id);
+            }
 
-            targetDocument.set(field.getName(), mapToDocumentDBValue(accessor.getProperty(property)));
+            return document;
+        } catch (JsonProcessingException e) {
```

Comment:
I suppose this exception can only be thrown by `objectMapper.writeValueAsString()`. Then I suggest minimize the scope of try clause. It's unclear to me which method will throw this.

Reply:
Will refine.